CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+# [Please contribute](#please-contribute)
+
+You can really make a difference by:
+
+- [Making an issue](https://help.github.com/articles/creating-an-issue/). A well-described issue helps a lot. (Check the [known issues](https://github.com/search?q=user%3Avitabaks+is%3Aissue+state%3Aopen) first.)
+- [Making a pull request](https://docs.github.com/fr/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) when you find an error in the code.
+
+I will try to help and take every contribution seriously.
+
+This is a great opportunity for me to learn how you use the role and also an opportunity for you to get into the habit of contributing to open-source software.
+
+## [Step by step](#step-by-step)
```

Comment:
There is another way, it is to use gitpod for development, which uses my [image](https://github.com/vitabaks/postgresql_cluster/blob/master/.gitpod.yml#L2) with already installed dependencies.

http://gitpod.io/#https://github.com/vitabaks/postgresql_cluster/pull/305 

Try it and maybe it's worth describing this way ?

Reply:
Indeed, using Gitpod seems to be a better option overall!

In any case, having a CONTRIBUTING.md file is important for guiding contributors.

Initially incorporating a Makefile (and possibly Taskfile later) is a good idea, as many open-source projects are built this way, and it allows people to understand the workflow more easily.

Another reason, which I find significant but is open to discussion, is that we're not restricted to a single dependency like Gitpod. The Makefile and virtual environment can work anywhere (VDI, Docker, Gitpod, etc.).

Please note, I'm a big fan of using Gitpod as well!

I will add the Gitpod section to the CONTRIBUTING.md file because it's a great addition.

I'll also test with your Docker image to see if I can incorporate the virtual environment directly. This way, through Gitpod, we can easily run a `make molecule-test` if needed.

What are your thoughts on this?

Reply:
I think I need to update the Dockerfile to install Python 3.10 and necessary tools by default. However, I would prefer to use Gitpod with the "tasks" method and initialize it via the makefile.

In summary:
- Remove the "# install molecule and ansible" section from Dockerfile.
- In .gitpod.yml, under tasks:
  - Under the "init" task, include the following commands:
     ```
     sudo apt update
     sudo apt install -y make python3.10 python3.10-venv
     python3 -m pip install --upgrade pip
     python3 -m pip install virtualenv
     ```
  - Under the "command" task, include `make reinitialization`.

This way, Gitpod will be consistent with the local setup (without Gitpod).

What do you think?

Reply:
the downside of this approach is the increase in the creation time of the gitpod working environment. 

Do we really need compliance with gitpod? 
Why can't we consider this as two independent options?

Reply:
I understand your concern about the increased creation time of the Gitpod working environment. However, the goal here is to provide a consistent experience for developers, whether they choose to work with Gitpod or on their local setup. This consistency can help reduce confusion and streamline the development process.

If we want to minimize the creation time, we should consider making similar optimizations to the Dockerfile, so that the same efficiency is achieved in both Gitpod and local setups. This way, developers will have the flexibility to choose their preferred environment without any major trade-offs in terms of setup time.

For example, we can update the Dockerfile to include Python 3.10 and the necessary tools by default, similar to what we did in the Gitpod configuration.

By doing this, we can maintain consistency between Gitpod and local environments, while also minimizing the creation time.

Let me know your thoughts on this.

Reply:
I would like to emphasize an important point for me. I believe it is crucial to maintain consistent functionality between Gitpod and local installations. Moreover, it is essential that people can quickly join the project and get started with ease.

Gitpod is an excellent choice, but I think it is important not to become too reliant on it. And what about those who need to be "offline" or on a private network?

That's why I think the Makefile approach, which would serve as a link, seems like a good choice.

Now, maybe you prefer that people learn to use Molecule and Python virtual environments? Perhaps it would be better to provide a basic Gitpod environment, such as the Ubuntu one you have, and add step-by-step installation instructions (the requirements.yml file remains essential to me). This way, new contributors can install in the same manner and have the same testing and contribution instructions, whether they start from the Gitpod base or their local machine.

What are your thoughts on this?

Reply:
good. If you have time, please provide a PR for gitpod based on how you see it.

But I think it is important that gitpod is already fully configured when starting the production environment.

It is necessary to save time, because it is the most valuable resource)